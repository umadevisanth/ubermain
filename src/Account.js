import { StyleSheet, View, Image, StatusBar } from 'react-native';
import Globalstyle from './Globalstyle';
import React, { useEffect } from 'react';
import { LinearGradient } from 'expo-linear-gradient';
import { useNavigation } from '@react-navigation/native';

const Account = (props) => {
  

  const navigation = useNavigation();

  useEffect(() => {
    const timer = setTimeout(() => {
      navigation.navigate('Login');
    }, 1500); //0.5s, adjust as needed

    return () => clearTimeout(timer); // Cleanup the timer on component unmount
  }, []);

  return (
    <View style={{ flex: 1 }}>
      <StatusBar backgroundColor='#000000' />
      <LinearGradient
        colors={['#1976D2', '#535AFF']}
        style={Globalstyle.main}>
        <View style={Globalstyle.accounttextview}>
          <View style={Globalstyle.Account_view}>
            <Image
              source={require('../assets/Grouplogo.png')}
              style={Globalstyle.accountviewimage}
            />
          </View>
        </View>
      </LinearGradient>
    </View>
  );
}

export default Account;

const styles = StyleSheet.create({});
