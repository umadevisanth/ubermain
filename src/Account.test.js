import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/native';
import Account from '../src/Account';
// Mock @expo/vector-icons
jest.mock('@expo/vector-icons', () => ({
  FontAwesome: 'FontAwesome', // Mock the specific icon component you're using
}));

// Mock useNavigation hook
jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));


describe('Account component', () => {
  it('renders correctly', () => {
    // Mock useNavigation behavior
    useNavigation.mockReturnValue({ navigate: jest.fn() });

    // Render the Login component
    const tree = renderer.create(<Account />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
