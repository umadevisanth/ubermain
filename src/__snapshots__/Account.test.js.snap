// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account component renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#276FDD",
      "height": "100%",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderRadius": 20,
          "borderWidth": 0,
          "elevation": 5,
          "height": 181,
          "top": 330,
          "width": 181,
        }
      }
    >
      <Text
        style={
          {
            "color": "#EDF6FF",
            "fontSize": 64,
            "paddingTop": 50,
          }
        }
      >
        Uber
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
